DevTools/Compiler/C++Compiler/LibCpp/Parser.cpp

```diff
@@ -284,8 +502,8 @@ NonnullRefPtr<Function> Parser::parse_function_definition()
     auto declarator = parse_declarator();
 
     if (declarator.has_value()) {
-        parse_function_body();
-        return create_ast_node<Function>(return_type, declarator.value().name, declarator.value().parameters);
+        auto body = parse_function_body();
+        return create_ast_node<Function>(return_type, declarator.value().name, declarator.value().parameters, body);
```

Comment:
maybe `move(body)`? not that important, just a nitpick

Reply:
The compiler says I can't
`cannot bind non-const lvalue reference of type 'AK::NonnullRefPtrVector<SIR::ASTNode>&' to an rvalue of type 'AK::NonnullRefPtrVector<SIR::ASTNode>`
I haven't used a lot of move on the parser

Reply:
Your constructor takes a `NonnullRefPtrVector<SIR::ASTNode>&` then copies it iirc, you should instead just take a `NonnullRefPtrVector<SIR::ASTNode>` and move the body from here.